tests/suites/test_suite_ssl.function

```diff
@@ -339,6 +341,259 @@ int mbedtls_mock_tcp_recv_nb( void *ctx, unsigned char *buf, size_t len )
     return mbedtls_test_buffer_get( socket->input, buf, len );
 }
 
+/*
+ * Structure with endpoint's certificates for SSL communication tests.
+ */
+typedef struct mbedtls_endpoint_certificate
+{
+    mbedtls_x509_crt ca_cert;
+    mbedtls_x509_crt cert;
+    mbedtls_x509_crt cert2;
+    mbedtls_pk_context pkey;
+    mbedtls_pk_context pkey2;
+} mbedtls_endpoint_certificate;
+
+/*
+ * Endpoint structure for SSL communication tests.
+ */
+typedef struct mbedtls_endpoint
+{
+        const char *name;
+        mbedtls_ssl_context ssl;
+        mbedtls_ssl_config conf;
+        mbedtls_ctr_drbg_context ctr_drbg;
+        mbedtls_entropy_context entropy;
+        mbedtls_mock_socket socket;
+        mbedtls_endpoint_certificate cert;
+} mbedtls_endpoint;
+
+#define EP( VAR )   ( &( ep -> VAR ) )
+#define CERT( VAR ) ( &( cert -> VAR ) )
+
+/*
+ * Initialize \p ep_cert structure and assigns it to endpoint
+ * represented by \p ep
+ *
+ * \retval  0 on success, otherwise error code
+ */
+int mbedtls_endpoint_certificate_init( mbedtls_endpoint *ep )
+{
+    int i = 0;
+    int ret = -1;
+    mbedtls_endpoint_certificate *cert;
+
+    if( ep == NULL )
+    {
+        return MBEDTLS_ERR_SSL_BAD_INPUT_DATA;
+    }
+
+    cert = &(ep->cert);
+    mbedtls_x509_crt_init( CERT(ca_cert) );
+    mbedtls_x509_crt_init( CERT(cert) );
+    mbedtls_x509_crt_init( CERT(cert2) );
+    mbedtls_pk_init( CERT(pkey) );
+    mbedtls_pk_init( CERT(pkey2) );
+
+    /* Load the trusted CA */
+
+    for( i = 0; mbedtls_test_cas[i] != NULL; i++ )
+    {
+        ret = mbedtls_x509_crt_parse( CERT(ca_cert),
+                        (const unsigned char*) mbedtls_test_cas[i],
+                        mbedtls_test_cas_len[i] );
+        TEST_ASSERT( ret == 0);
+    }
+
+    for( i = 0; mbedtls_test_cas_der[i] != NULL; i++ )
+    {
+        ret = mbedtls_x509_crt_parse_der( CERT(ca_cert),
+                        (const unsigned char*) mbedtls_test_cas_der[i],
+                        mbedtls_test_cas_der_len[i] );
+        TEST_ASSERT( ret == 0);
+    }
+
+    /* Load own certificate and private key */
+
+    if( ep->conf.endpoint == MBEDTLS_SSL_IS_SERVER )
+    {
+        ret = mbedtls_x509_crt_parse( CERT(cert),
+                            (const unsigned char*) mbedtls_test_srv_crt_rsa,
+                            mbedtls_test_srv_crt_rsa_len );
+        TEST_ASSERT( ret == 0);
+
+        ret = mbedtls_pk_parse_key( CERT(pkey),
+                            (const unsigned char*) mbedtls_test_srv_key_rsa,
```

Comment:
This will be probably parametrized later on to extend the test coverage, but for now it's ok.